### ISSUE TITLE
Console: refresh is sometimes needed after starting build

### ISSUE THREAD

**Issue by pmorie** [openshift/origin#2813]
Sometimes when you trigger a build from the console, the UI doesn't show the build and you have to refresh in order to see it.

**Comment by pmorie** [comment]
@liggitt

**Comment by liggitt** [comment]
recreatable? steps?

**Comment by pmorie** [comment]
Create a new project with build, click 'create build'.  ~30% of the time, you'll hit this bug.

**Comment by spadgett** [comment]
@pmorie if you hit this again, please check if there are any errors in the browser's Javascript console.

**Comment by spadgett** [comment]
@pmorie Is the build not showing on the overview page or the browse builds page (or both)?

**Comment by ryanj** [comment]
I'm hitting this issue with Firefox on Fedora21.  The green alert message seems to appear reliably, but these two UI triggers seem less reliable:

* The 'Build' button should automatically change state to 'disabled' whenever a build is in progress.
* A build's progress section (displaying total build time) should be displayed whenever a build is in progress.

I've also seen the timer reset back to zero after the first 20 or 30 seconds of the build, which means the total build time may end up taking 20-30s longer than reported.

**Comment by pmorie** [comment]
@spadgett this is scoped to the builds page from what I can tell but I was not especially vigilant about the other views that might be displaying a build's info / pod.

**Comment by liggitt** [comment]
The `Build` button should disable as soon as the server notifies the UI of a build... if the UI is displaying a build in progress and the button is still enabled, that's a bug. Otherwise, the UI hasn't been notified yet.

The timestamp resets when the build changes state:
```
              <span ng-switch-when="Running">running for <duration-until-now timestamp="build.status.startTimestamp"></duration-until-now></span>
              <span ng-switch-when="New">waiting for <duration-until-now timestamp="build.metadata.creationTimestamp"></duration-until-now></span>
              <span ng-switch-when="Pending">waiting for <duration-until-now timestamp="build.metadata.creationTimestamp"></duration-until-now></span>
```

**Comment by spadgett** [comment]
After talking to @ryanj, I believe this is the same problem as #2529.

**Comment by detiber** [comment]
@spadgett We've verified and the fix for https://github.com/openshift/origin/issues/2529 is deployed in the environment. Could it be related to https://github.com/openshift/origin/pull/2839?

**Comment by liggitt** [comment]
#2839 prevented builds from appearing on the overview page, but not on the builds page.

**Comment by spadgett** [comment]
@detiber Just confirming the fix for #2529 was there when the problem was observed? I'm not sure how often the server is updated, but we couldn't reproduce yesterday.

Your server is still using v1beta1 OpenShift API, so the change in #2839 would not apply to you.
